pkg/controllers/vdb/init_db.go

```diff
@@ -99,6 +99,16 @@ func (g *GenericDatabaseInitializer) runInit(ctx context.Context) (ctrl.Result,
 		return res, err
 	}
 
+	sec := vapi.MakeNMATLSSecretRef(g.Vdb.Spec.NMATLSSecret)
+	if err := vdbstatus.UpdateSecretRef(ctx, g.VRec.GetClient(), g.Vdb, sec); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	clientSec := vapi.MakeClientServerTLSSecretRef(g.Vdb.Spec.ClientServerTLSSecret)
+	if err := vdbstatus.UpdateSecretRef(ctx, g.VRec.GetClient(), g.Vdb, clientSec); err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
Instead of calling k8s multiple times, you can add a new function that get a slice of secretRefs. See `vrpqstatus/status.go/Update()` as an example.